### PR TITLE
Patching and retrieving node reports in host admin.

### PR DESCRIPTION
**Pull request by hakonhall**
- A report can be patched to the node repository by using NodeAttributes and
   NodeRepository.updateNodeAttributes().
 - A report can be retrieved as a Jackson class from NodeSpec and its
   NodeReports.

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeReports.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    public NodeReports(NodeReports reports) {
+        this.reports.putAll(reports.reports);
+    }
+
+    private NodeReports(Map<String, JsonNode> reports) {
+        this.reports.putAll(reports);
+    }
+
+    public static NodeReports fromMap(Optional<Map<String, JsonNode>> reports) {
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeReports.java]: This `Optional` check should be done by the caller, most of the users make a redundant wrap in `Optional`...

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeSpec.java [hunk 2]
```diff
@@ -117,6 +121,7 @@ public NodeSpec(
         this.ipAddresses = Objects.requireNonNull(ipAddresses);
         this.hardwareDivergence = Objects.requireNonNull(hardwareDivergence);
         this.hardwareFailureDescription = Objects.requireNonNull(hardwareFailureDescription);
+        this.reports = reports;
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeSpec.java]: `requireNonNull`?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/reports/BaseReport.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.configserver.reports;
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/reports/BaseReport.java]: Shouldn't this be under `com.yahoo.vespa.hosted.node.admin.configserver.noderepository.bindings`?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeSpec.java (hunk 2 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeSpec.java]: Fixed

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/reports/BaseReport.java (hunk 3 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/reports/BaseReport.java]: Moved to under noderepository.

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeReports.java (hunk 1 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/NodeReports.java]: Fixed

**Review by freva — APPROVED** [review]

**Pull request merged**